### PR TITLE
Add Alternative Email handlers

### LINKED ISSUE #607: [FEATURE] Remove sendgrid dependency
### Is there an existing issue or pull request for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe it

Currently when setting up a self-hosted instance, users must create a sendgrid api key

### Describe the solution you'd like

Be able to use SMTP as well, to reduce cloud dependency

### Describe alternatives you've considered

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by cherry-john**
### Description
API Email services are great until they stop allowing signups or their servers become untrusted.
This PR adds a variety of other API Email services, alongside SMTP for direct sending. Whilst API Email services are generally more reputable and preferred, SMTP makes sense to add too.

#### Providers Implemented
- [x] Mailgun
- [x] Postmark
- [x] SMTP

Closes #607 
Documented in adam-rms/website#76

### Checklist

- [x] I accept the contributor license agreement for this repository.
- [x] I have added documentation for new/changed functionality in this PR or in the [documentation repo](https://github.com/adam-rms/website).
- [x] I have updated the API documentation for any routes changed, within each api file.
- [x] All active GitHub checks for tests, formatting, and security are passing.
- [x] The correct base branch is being used (if not `main`).
- [x] A GitHub issue is linked to this pull request.

### CODE REVIEW

File: src/common/libs/Config/configStructureArray.php [hunk 1]
```diff
@@ -130,6 +172,27 @@
     "default" => false,
     "envFallback" => "bCMS__FROM_EMAIL",
   ],
+  "EMAILS_FROMDOMAIN" => [
```
Comment (Jbithell) [inline src/common/libs/Config/configStructureArray.php:175]: Could we not just grab this from the from email?

Comment (Jbithell) [reply src/common/libs/Config/configStructureArray.php:175]: Or is it like a config thing like domain id 182627

**Comment by Jbithell** [comment]
I came across this which is cool but python only https://github.com/caronc/apprise

File: src/common/libs/Config/configStructureArray.php (hunk 1 above)
Comment (cherry-john) [reply src/common/libs/Config/configStructureArray.php:175]: Thought it was specific, turns out we can!

**Comment by Jbithell** [comment]
😍😍😍

**Review by Jbithell — APPROVED** [review]

**Pull request merged**